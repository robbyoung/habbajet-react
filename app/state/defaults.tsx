import {HabbajetEditor, PurchaseEditor} from '.';
import {habbajetColors} from '../colors';

const EMPTY_FIELD = {
    value: '',
    errorMessage: '',
};

export const DEFAULT_EDITOR_STATE: HabbajetEditor = {
    name: EMPTY_FIELD,
    value: EMPTY_FIELD,
    modifier: EMPTY_FIELD,
    slack: EMPTY_FIELD,
    color: habbajetColors[0],
    validated: false,
};

export const DEFAULT_PURCHASE_EDITOR_STATE: PurchaseEditor = {
    name: EMPTY_FIELD,
    cost: EMPTY_FIELD,
    tagId: '',
    validated: false,
};
