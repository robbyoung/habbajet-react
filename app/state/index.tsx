export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
