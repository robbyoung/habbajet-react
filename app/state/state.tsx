export interface Habbajet {
    name: string;
    value: number;
    days: boolean[];
}

export interface Purchase {
    name: string;
    value: number;
}

export interface State {
    habits: Habbajet;
    purchages: Purchase;
    budget: string;
}
