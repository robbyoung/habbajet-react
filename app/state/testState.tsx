import {
    State,
    Habbajet,
    Purchase,
    HabbajetEditor,
    Tag,
    PurchaseEditor,
    TagEditor,
} from '.';
import {habbajetColors} from '../colors';

export function createTestState(
    numHabbajets: number,
    numPurchases: number,
    budget: number,
    selectedIndex: number = -1,
): State {
    const habbajets: Habbajet[] = [];
    for (let i = 0; i < numHabbajets; i++) {
        habbajets.push({
            id: `${i}`,
            name: `Habbajet ${i}`,
            maxValue: 100,
            currentValue: 100,
            modifier: 2,
            results: [],
            color: habbajetColors[i % habbajetColors.length],
            date: '2020-03-22T11:00:00.000Z',
            toClaim: false,
            selected: i === selectedIndex,
            currentStreak: 0,
            bestStreak: 5,
            oldStreaks: [0, 5],
            totalSlack: 0,
            remainingSlack: 0,
            description: '',
        });
    }

    const purchases: Purchase[] = [];
    for (let i = 0; i < numPurchases; i++) {
        purchases.push({
            id: `${i}`,
            name: `Purchase ${i}`,
            cost: i,
            date: '2020-03-22T11:00:00.000Z',
            tagId: `${i % 10}`,
        });
    }

    const tags: Tag[] = [];
    for (let i = 0; i < 10; i++) {
        tags.push({
            id: `${i}`,
            name: `Tag ${i}`,
            color: habbajetColors[i % habbajetColors.length],
        });
    }

    const EMPTY_FIELD = {
        value: '',
        errorMessage: '',
    };
    const habbajetEditor: HabbajetEditor = {
        name: {...EMPTY_FIELD},
        description: {...EMPTY_FIELD},
        value: {...EMPTY_FIELD},
        modifier: {...EMPTY_FIELD},
        slack: {...EMPTY_FIELD},
        color: habbajetColors[0],
        validated: false,
    };

    const purchaseEditor: PurchaseEditor = {
        name: {...EMPTY_FIELD},
        cost: {...EMPTY_FIELD},
        tagId: '',
        validated: false,
    };

    const tagEditor: TagEditor = {
        name: {...EMPTY_FIELD},
        color: habbajetColors[0],
        validated: false,
    };

    return {
        habbajets,
        purchases,
        budget,
        habbajetEditor,
        purchaseEditor,
        tagEditor,
        tags,
    };
}

export function createTestEditor(
    name: string,
    value: string,
    modifier: string,
    slack = '0',
    validated = false,
): HabbajetEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        description: {
            value: '',
            errorMessage: '',
        },
        value: {
            value,
            errorMessage: '',
        },
        modifier: {
            value: modifier,
            errorMessage: '',
        },
        slack: {
            value: slack,
            errorMessage: '',
        },
        color: habbajetColors[0],
        validated,
    };
}

export function createTestPurchaseEditor(
    name: string,
    value: string,
    validated = false,
): PurchaseEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        cost: {
            value,
            errorMessage: '',
        },
        tagId: '0',
        validated,
    };
}

export function createTestTagEditor(
    name: string,
    validated = false,
): TagEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        color: '#000000',
        validated,
    };
}
