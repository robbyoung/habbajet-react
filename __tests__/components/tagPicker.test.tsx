import React from 'react';
import renderer from 'react-test-renderer';
import TagPicker from '../../app/components/tagPicker';
import {createTestState} from '../../app/state/testState';
import {render, fireEvent, wait} from '@testing-library/react-native';

describe('TagPicker Component', () => {
    it('can render a tag picker', () => {
        const tags = createTestState(0, 0, 0).tags;
        const component = renderer.create(
            <TagPicker
                tags={tags}
                selected="0"
                onNewTag={() => undefined}
                onSelect={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('can render a tag picker with no selected tag', () => {
        const tags = createTestState(0, 0, 0).tags;
        const component = renderer.create(
            <TagPicker
                tags={tags}
                selected=""
                onNewTag={() => undefined}
                onSelect={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onSelect callback if a tag is selected', async () => {
        const tags = createTestState(0, 0, 0).tags;
        const onSelect = jest.fn();
        const {getByTestId} = render(
            <TagPicker
                tags={tags}
                selected=""
                onSelect={onSelect}
                onNewTag={() => undefined}
            />,
        );

        const tagToSelect = getByTestId(
            `${tags[0].name.replace(' ', '-')}-tag`,
        );
        fireEvent.press(tagToSelect);
        await wait(() => expect(onSelect).toBeCalledWith(tags[0].id));
    });

    it('will run the onNewTag callback if the new tag button is selected', async () => {
        const tags = createTestState(0, 0, 0).tags;
        const onCreate = jest.fn();
        const {getByTestId} = render(
            <TagPicker
                tags={tags}
                selected=""
                onSelect={() => undefined}
                onNewTag={onCreate}
            />,
        );

        const newTagButton = getByTestId('new-tag');
        fireEvent.press(newTagButton);
        await wait(() => expect(onCreate).toBeCalled());
    });
});
