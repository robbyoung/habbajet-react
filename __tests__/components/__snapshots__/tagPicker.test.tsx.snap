// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagPicker Component can render a tag picker 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#959595",
          "fontFamily": "Abel",
          "fontSize": 20,
          "margin": 0,
          "padding": 0,
        },
      ]
    }
  >
    Tag
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#8066C2",
          "borderColor": "#959595",
          "borderRadius": 4,
          "borderStyle": "dotted",
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 0
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E04747",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2AAD5E",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 2
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#B1B952",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 3
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#7F90EC",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 4
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#37353C",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 5
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E64CE9",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 6
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#EB712D",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 7
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#8066C2",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 8
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E04747",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 9
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#959595",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        New Tag
      </Text>
    </View>
  </View>
</View>
`;

exports[`TagPicker Component can render a tag picker with no selected tag 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#959595",
          "fontFamily": "Abel",
          "fontSize": 20,
          "margin": 0,
          "padding": 0,
        },
      ]
    }
  >
    Tag
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#8066C2",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 0
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E04747",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2AAD5E",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 2
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#B1B952",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 3
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#7F90EC",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 4
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#37353C",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 5
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E64CE9",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 6
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#EB712D",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 7
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#8066C2",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 8
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E04747",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        Tag 9
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#959595",
          "borderColor": "transparent",
          "borderRadius": 8,
          "borderWidth": 4,
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "padding": 5,
          }
        }
      >
        New Tag
      </Text>
    </View>
  </View>
</View>
`;
